src/modules/communication/grpc_util.cpp

```diff
@@ -1,41 +1,30 @@
 #include "grpc_util.hpp"
 #include "../../../include/json_schema.hpp"
 #include "msg_schema.hpp"
+#include <cstring>
 namespace gruut{
-    std::string HeaderController::attachHeader(std::string &compressed_json, MessageType message_type) {
+    std::string HeaderController::attachHeader(std::string &compressed_json,
+                                               MessageType msg_type, MACAlgorithmType mac_algo_type, CompressionAlgorithmType compression_algo_type) {
         std::string header;
         uint32_t total_length = static_cast<uint32_t>(HEADER_LENGTH + compressed_json.size());
 
         header.resize(32);
         header[0] = G;
         header[1] = VERSION;
-        header[2] = static_cast<uint8_t>(message_type);
-        header[3] = MAC;
-        header[4] = COMPRESSION_TYPE;
+        header[2] = static_cast<uint8_t>(msg_type);
```

Comment:
tests/modules/test.cpp 처럼 `header[0]` 으로 header를 생성하는 것보다 `MessageHeader` 구조체의 데이터를 채워나가는게 더 가독성이 좋지 않을까요? 고치는건 @blackironj 재량입니다.

```suggestion
        MessageHeader msg_hdr;
        msg_hdr.identifier = 'G';
        msg_hdr.version = '1';
```

Reply:
제가 봤을땐 `MessageHeader`의 구조가 변경되면 이곳에서 버그가 발생할 수 있을 것 같습니다.

Reply:
@park53kr 헤더를 만들 때 structure 에 넣는다 하여도 다시 bytes string으로 변환하는 작업이 필요하다고 생각합니다.  structure를 한번에 그냥 serialize 하는것도 생각을 해봤는데, 자바쪽에서 해놓으신 코드가 저 처럼 바이트단위로 파싱합니다. 지금은 이렇게 하는게 나을것 같습니다. 헤더가 바뀌면 이쪽 코드도 다시 고쳐야 할것 같습니다. 현재 헤더에 6bytes 하나가 있어서 애매해졌습니다.